testsuite/features/step_definitions/command_steps.rb

```diff
@@ -179,6 +179,17 @@
   $command_output, _code = get_target('server').run(command)
 end
 
+When(/^I use spacewalk-common-channel to add all "([^"]*)" channels with arch "([^"]*)"$/) do |channel, architecture|
+  channels_to_synchronize = CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION.dig(product, "#{channel}-#{architecture}")
```

Comment:
what does `.dig` bring with respect to `[][]` ?

maybe we should be a bit conservative here and stay aligned with the other similar uses of  `CHANNEL_TO_SYNC_BY_OS_PRODUCT_VERSION` which use the sequence of square brackets

Reply:
The `dig` method is useful for safely navigating through nested hashes without raising an error if any intermediate key is `nil`. It returns `nil` if any key in the chain is not found.

This way we exactly know what the type of `channels_to_synchronize` would be (a `list` or `nil`), what makes it more predictable in further stages if we do not have a confidence that the keys exist at each level of our dictionary.

Reply:
there's not much risk at first level ("SUSE Manager" or "Uyuni").

but ok